WordPress/gutenberg
# Block Supports: Switch dimensions inspector controls slot to bubble virtually

Related: 
- https://github.com/WordPress/gutenberg/pull/34157
- https://github.com/WordPress/gutenberg/issues/34766

## Description

Updates the Dimensions InspectorControls slot group to bubble virtually. 

- Adds wrapper to Slot that retrieves `ToolsPanelContext` and passes that through `fillProps`
- Removes unused `header` prop from `BlockSupportToolsPanel`
- Updates the InspectorControls fills to render via a function. This allows recreation of `ToolsPanelContext` provider.
- Added new styles to `ToolsPanel` component to handle injected `div` by the virtual bubbling `Slot`.

_Note: The virtually bubbling slot requires a non-functional component to apply refs to. So far I haven't found a way around this `div`. As a result, this PR updates the `ToolsPanel` styling such that it will follow the same CSS grid layout as the panel itself._

**It appears this switch to virtual bubbling will also resolve a recently discovered bug with support controls which are only enabled via block specific settings within theme.json. See #34766**

## How has this been tested?

1. Edit a post, add a group block and select it.
2. Confirm that the Dimensions panel is correctly displayed in the Inspector Controls.
3. Test that editing the padding field and resetting/toggling it via the tools panel menu works as expected.
4. Opt into `margin` support via the group `block.json` file.
5. Enable the margin support UI via a **block specific context** within your `theme.json`
```
"settings": {
   ...
   "spacing": {
	"customMargin": false,
	"customPadding": false,
	"units": [ "px", "em", "rem", "vh", "vw", "%" ]
   },
    ...
    "blocks": {
        "core/group": {
		"spacing": {
			"customMargin": true,
			"customPadding": true
			}
		},
```
6. Reload editor and confirm both margin and padding support controls are present, display correctly and work as expected.

## Types of changes
Enhancement. Bug fix.

## Checklist:
